### ISSUE TITLE
Rule: no-fx-direct-call

### ISSUE THREAD

**Issue by binjospookie** [effector/eslint-plugin#47]
// 👍
const myFx = createEffect();
const evt = createEvent();

forward({
  from: createEvent,
  to: myFx
})

// 👎
const myFx = createEffect();

myFx()

**Comment by igorkamyshev** [comment]
Could you describe your idea, please?

In my opinion, this case should be covered by https://github.com/effector/eslint-plugin/issues/30

**Comment by binjospookie** [comment]
In my opinion calling the effect directly gives less control over its calls. By calling with events we have the possibility to describe the conditions for calling the effect explicitly and not smeared in the application.

For example:
```ts
const evt = createEvent();
const someFx = createEffect();

// today
forward({
  from: evt,
  to: someFx
});

// maybe tomorrow, without changing whole app
sample({
  clock: guard({
    clock: evt,
    source: someFx.pending,
    filter: (isPending) => !isPending
  }),
  source: evt,
  target: someFx
});
```

**Comment by igorkamyshev** [comment]
OK, I see.

Let me think about it a couple of times. And maybe someone from @effector/core will provide their opinion of this topic.

**Comment by popuguytheparrot** [comment]
What about

```js
const someFx = createEffect();
const evt = someFx.prepend()
```

**Comment by AlexandrHoroshih** [comment]
What about effect calls inside other effects? It is allowed and perfectly valid, [if only effects are awaited](https://effector.dev/docs/api/effector/scope#imperative-effects-calls-with-scope)

```ts
const controllerFx = createEffect(async () => {
 const result = await oneFx()
 return await twoFx(result)
})
```
Maybe, even separate rule is needed for that 🤔

This feature, usually, not very useful for main logic of the app, but is very handy, when creating operators like [this](https://github.com/effector/patronum/discussions/167)

**Comment by sergeysova** [comment]
@AlexandrHoroshih I propose "Effects called inside another effect should not be warned."

Cause I can await it soon:

```ts
const publishReleaseFx = createEffect(async () => {
  await Promise.all([
    authUserFx(),
    pushCommitFx(),
    buildCommitFx(),
    pushBuildFx(),
  ])
})
```
